core/src/main/java/org/elasticsearch/node/NodeModule.java

```diff
@@ -58,5 +97,30 @@ protected void configure() {
         bind(Node.class).toInstance(node);
         bind(MonitorService.class).toInstance(monitorService);
         bind(NodeService.class).asEagerSingleton();
+        bind(ProcessorsRegistry.class).toInstance(processorsRegistry);
+    }
+
+    /**
+     * Returns the node
+     */
+    public Node getNode() {
+        return node;
+    }
+
+    /**
+     * Adds a processor factory under a specific type name.
+     */
+    public void registerProcessor(String type, Function<TemplateService, Processor.Factory<?>> processorFactoryProvider) {
+        processorsRegistry.registerProcessor(type, processorFactoryProvider);
+    }
+
+    public static boolean isNodeIngestEnabled(Settings settings) {
+        return settings.getAsBoolean("node.ingest", true);
+    }
+
+    public static boolean isNodeIngestEnabled(ImmutableOpenMap<String, String> nodeAttributes) {
+        String ingestEnabled = nodeAttributes.get("ingest");
+        //reproduces same logic used in settings.getAsBoolean used above
+        return Booleans.parseBoolean(ingestEnabled, true);
```

Comment:
read the above comment, we can't have this method return something that's not inline with the other isNodeIngestEnabled. so we can't make this change unless we have a settings.getExactBoolean, sorry it's a PITA I know. We should get rid of those parseBoolean lenient methods. Maybe also keeping a list of ingestNodes in the cluster state would remove the need for this method.